### PR TITLE
fix: Remove EC2-classic deprecation warnings by hardcoding classiclink values to `null`

### LINKED ISSUE #824: Deprecate vpc_classic_link
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: 4.28.0

## Is your request related to a problem? Please describe.

```
╷
│ Warning: Argument is deprecated
│ 
│   with module.vpc.aws_vpc.this,
│   on .terraform/modules/vpc/main.tf line 27, in resource "aws_vpc" "this":
│   27:   enable_classiclink               = var.enable_classiclink
│ 
│ With the retirement of EC2-Classic the enable_classiclink attribute has
│ been deprecated and will be removed in a future version.
│ 
│ (and 3 more similar warnings elsewhere)
╵
```

## Describe the solution you'd like.

Implement a migration path once the classiclink is completely removed from the AWS provider.

## Additional context

Warning shows up since we bumped the AWS provider to 4.28.0.

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Remove EC2-classic deprecation warnings by hardcoding classiclink values to `null`
	- Due to https://github.com/hashicorp/terraform/issues/31730 - defaulting to `null` does not resolve the warning and instead we have to hardcode the value as `null` to satisfy. Users cannot create new EC2-classic VPCs today, and users who are still using EC2-classic are required to upgrade (per AWS) - users who are using EC2-classic created under this module should pin their module version to v3.14.3 until they have migrated their VPC out of EC2-classic

## Motivation and Context
- Avoid spamming users with warnings
- Closes #824 

## Breaking Changes
- Per instructions above, not marking as breaking since this is AWS's side. Once users migrate their VPC out of EC2-classic (required by AWS), users can continue using v3.14.3+ again

tl;dr - the breaking change is on the AWS side, not the module

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -694,8 +696,9 @@ variable "default_vpc_enable_dns_hostnames" {
   default     = false
 }
 
+# tflint-ignore: terraform_unused_declarations
```
Comment (antonbabenko) [inline variables.tf:699]: Handy hint to tflint :)

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 3.14.4 :tada:
